code/datums/ai/monkey/monkey_controller.dm

```diff
@@ -103,42 +113,142 @@ have ways of interacting with a specific mob and control it.
 			continue
-		W = i
-		break
+		pickpocket = TRUE
+		weapon = held_weapon
+
+	if(!weapon || (weapon in living_pawn.held_items))
+		return FALSE
 
-	if(W)
-		blackboard[BB_MONKEY_PICKUPTARGET] = W
-		current_movement_target = W
+	blackboard[BB_MONKEY_PICKUPTARGET] = weapon
+	current_movement_target = weapon
+	if(pickpocket)
+		current_behaviors += GET_AI_BEHAVIOR(/datum/ai_behavior/monkey_equip/pickpocket)
+	else
 		current_behaviors += GET_AI_BEHAVIOR(/datum/ai_behavior/monkey_equip/ground)
+	return TRUE
+
+/// Returns either the best weapon from the given choices or null if held weapons are better
+/datum/ai_controller/monkey/proc/GetBestWeapon(list/choices, list/held_weapons)
+	var/gun_neurons_activated = blackboard[BB_MONKEY_GUN_NEURONS_ACTIVATED]
+	var/top_force = 0
+	var/obj/item/top_force_item
+	for(var/obj/item/item as anything in held_weapons)
+		if(!item)
+			continue
+		if(HAS_TRAIT(item, TRAIT_NEEDS_TWO_HANDS) || blackboard[BB_MONKEY_BLACKLISTITEMS][item])
+			continue
+		if(gun_neurons_activated && istype(item, /obj/item/gun))
+			// We have a gun, why bother looking for something inferior
+			// Also yes it is intentional that monkeys dont know how to pick the best gun
+			return item
+		if(item.force > top_force)
+			top_force = item.force
+			top_force_item = item
+
+	for(var/obj/item/item as anything in choices)
+		if(!item)
+			continue
+		if(HAS_TRAIT(item, TRAIT_NEEDS_TWO_HANDS) || blackboard[BB_MONKEY_BLACKLISTITEMS][item])
+			continue
+		if(gun_neurons_activated && istype(item, /obj/item/gun))
+			return item
+		if(item.force <= top_force)
+			continue
+		top_force_item = item
+		top_force = item.force
+
+	return top_force_item
+
+/datum/ai_controller/monkey/proc/TryFindFood()
+	. = FALSE
+	var/mob/living/living_pawn = pawn
+
+	// Held items
+
+	var/list/food_candidates = list()
+	for(var/obj/item as anything in living_pawn.held_items)
+		if(!item || !IsEdible(item))
+			continue
+		food_candidates += item
+
+	for(var/obj/item/candidate in oview(2, living_pawn))
+		if(!IsEdible(candidate))
+			continue
+		food_candidates += candidate
+
+	if(length(food_candidates))
+		var/obj/item/best_held = GetBestWeapon(null, living_pawn.held_items)
+		for(var/obj/item/held as anything in living_pawn.held_items)
+			if(!held || held == best_held)
+				continue
+			living_pawn.dropItemToGround(held)
+
+		AddBehavior(/datum/ai_behavior/consume, pick(food_candidates))
 		return TRUE
-	else
-		var/mob/living/carbon/human/H = locate(/mob/living/carbon/human/) in oview(2,living_pawn)
-		if(H)
-			W = pick(H.held_items)
-			if(W && !blackboard[BB_MONKEY_BLACKLISTITEMS][W] && W.force > blackboard[BB_MONKEY_BEST_FORCE_FOUND] && !HAS_TRAIT(W, TRAIT_NEEDS_TWO_HANDS))
-				blackboard[BB_MONKEY_PICKUPTARGET] = W
-				current_movement_target = W
-				current_behaviors += GET_AI_BEHAVIOR(/datum/ai_behavior/monkey_equip/pickpocket)
-				return TRUE
+
+/datum/ai_controller/monkey/proc/IsEdible(obj/item/thing)
```

Comment:
this could probably be a global helper

Reply:
also this will work on empty glasses which could get the monkey stuck maybe?

Reply:
The monkey wont get stuck but yeah I'll put an empty check in here